applications/ContactStructuralMechanicsApplication/custom_conditions/ALM_frictionless_components_mortar_contact_condition.cpp

```diff
@@ -80,12 +81,12 @@ void AugmentedLagrangianMethodFrictionlessComponentsMortarContactCondition<2,2,
     const bounded_matrix<double, 2, 2>& X1 = rDerivativeData.X1;
     const bounded_matrix<double, 2, 2>& X2 = rDerivativeData.X2;
     
-    const bounded_matrix<double, 2, 2>& LM = MortarUtilities::GetVariableMatrix<2, 2>(this->GetGeometry(), VECTOR_LAGRANGE_MULTIPLIER, 0);
+    const bounded_matrix<double, 2, 2> LM = MortarUtilities::GetVariableMatrix<2, 2>(this->GetGeometry(), VECTOR_LAGRANGE_MULTIPLIER, 0);
     
     const bounded_matrix<double, 2, 2>& NormalSlave = rDerivativeData.NormalSlave;
 
     // The ALM parameters
-    const array_1d<double, 2>& DynamicFactor = MortarUtilities::GetVariableVector<2>(this->GetGeometry(), DYNAMIC_FACTOR);
+    const array_1d<double, 2> DynamicFactor = MortarUtilities::GetVariableVector<2>(this->GetGeometry(), DYNAMIC_FACTOR);
```

Comment:
here the & was correct

Reply:
take into account I am building an auxiliary matrix, is not directly the FastGetSolutionStepValue

Reply:
well ok, you may need it initialized. anyway...your call